### PR TITLE
DOC: Add tuple description to allowed inputs for iloc #47799

### LINKED ISSUE #47799: DOC: iloc docs should indicate a tuple is valid
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.iloc.html

### Documentation problem

In the docs, we say these are valid parameters:

- An integer, e.g. 5.
- A list or array of integers, e.g. [4, 3, 0].
- A slice object with ints, e.g. 1:7.
- A boolean array.
- A callable function with one argument (the calling Series or DataFrame) and that returns valid output for indexing (one of the above). This is useful in method chains, when you don’t have a reference to the calling object, but would like to base your selection on some value.

But a 2D tuple is valid as well, as shown in the example at the bottom of the page.  So this should be indicated in the docs.

Also, a 1D tuple like `df.iloc[0:2,]` works as well.

See discussion in pandas-stubs at https://github.com/pandas-dev/pandas-stubs/pull/152

### Suggested fix for documentation

Add tuple description to list at the top.

### PR DESCRIPTION
**Pull request by pratikpatel2512**
- [x] closes #47799 
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

File: pandas/core/indexing.py [hunk 1]
```diff
@@ -155,6 +155,8 @@ def iloc(self) -> _iLocIndexer:
           DataFrame) and that returns valid output for indexing (one of the above).
           This is useful in method chains, when you don't have a reference to the
           calling object, but would like to base your selection on some value.
+        - A tuple of row and column indexes, e.g. ``0, 1`` or ``[1, 2], [2, 5]`` or
```
Comment (phofl) [inline pandas/core/indexing.py:158]: Maybe add that the tuple elements consist of one of the above mentioned ones instead of given examples

Comment (attack68) [reply pandas/core/indexing.py:158]: Although I would add the parentheses to make it explit they are tuples. e,g, ``([1,2], [2,5])``.

**Comment by pratikpatel2512** [comment]
Acknowledged the comments. Updated the statement.

File: pandas/core/indexing.py (hunk 1 above)
Comment (pratikpatel2512) [reply pandas/core/indexing.py]: Acknowledged the comments and updated the statement.

Comment (pratikpatel2512) [reply pandas/core/indexing.py]: Are we suppose to run around 30 some checks for document updates?

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @pratikpatel2512

**Pull request merged**